### ISSUE TITLE
Error using BatchInterceptor with browserInterceptors in browser environment.

### ISSUE THREAD

**Issue by steve-zhu-sh** [mswjs/interceptors#476]
Still getting the issue from here: https://github.com/mswjs/interceptors/pull/361#issuecomment-1455223410

```
Type 'readonly [FetchInterceptor, XMLHttpRequestInterceptor]' is not assignable to type 'readonly Interceptor<any>[]'.
  Type 'FetchInterceptor | XMLHttpRequestInterceptor' is not assignable to type 'Interceptor<any>'.
    Type 'FetchInterceptor' is not assignable to type 'Interceptor<any>'.
      Types have separate declarations of a private property 'symbol'.ts(2322)
```

```typescript
import { BatchInterceptor } from '@mswjs/interceptors';
import browserInterceptors from '@mswjs/interceptors/presets/browser';

const interceptor = new BatchInterceptor({
  name: 'interceptor',
  interceptors: browserInterceptors,
});
```

This looks to be because the exports in package.json for "." does not include a browser config.
It should be something like this:
```json
{
 // ...
  "exports": {
    ".": {
      "browser": {
        "types": "./lib/browser/index.d.ts",
        "require": "./lib/browser/index.js",
        "import": "./lib/browser/index.mjs",
        "default": "./lib/browser/index.js"
      },
      "types": "./lib/node/index.d.ts",
      "require": "./lib/node/index.js",
      "import": "./lib/node/index.mjs",
      "default": "./lib/node/index.js"
    },
    // ...
  },
  // ...
}

**Comment by kettanaito** [comment]
Hey, @steve-zhu-sh. Thanks for reporting this. 

This is strange. We _do_ have the `browser` export for the `/presets/browser` path:

- [(Modern) Export condition](https://github.com/mswjs/interceptors/blob/3258f5cc62c472ba63f5c70e455d5b245c74a216/package.json#L60-L61)
- [CommonJS placeholder](https://github.com/mswjs/interceptors/blob/3258f5cc62c472ba63f5c70e455d5b245c74a216/presets/browser/package.json#L4)

Can you please make sure you're using the latest version of `@mswjs/interceptors`? Perhaps a version mismatch issue. If not, a reproduction repository would be the way forward to resolve this. Thanks.

**Comment by steve-zhu-sh** [comment]
Hi @kettanaito , the `/preset/browser` path does have the `browser` export, but the `.` path also needs a `browser` export, otherwise the definition for `BatchInterceptor` is not compatible with `browserInterceptors`. Since `BatchInterceptor` is imported from the `.` path.

**Comment by steve-zhu-sh** [comment]
Here's the entire exports change needed for clarity:

```diff
  "exports": {
    ".": {
+     "browser": {
+       "types": "./lib/browser/index.d.ts",
+       "require": "./lib/browser/index.js",
+       "import": "./lib/browser/index.mjs",
+       "default": "./lib/browser/index.js"
+     },
      "types": "./lib/node/index.d.ts",
      "require": "./lib/node/index.js",
      "import": "./lib/node/index.mjs",
      "default": "./lib/node/index.js"
    },
    "./ClientRequest": {
      "browser": null,
      "types": "./lib/node/interceptors/ClientRequest/index.d.ts",
      "require": "./lib/node/interceptors/ClientRequest/index.js",
      "import": "./lib/node/interceptors/ClientRequest/index.mjs",
      "default": "./lib/node/interceptors/ClientRequest/index.js"
    },
    "./XMLHttpRequest": {
      "browser": {
        "types": "./lib/browser/interceptors/XMLHttpRequest/index.d.ts",
        "require": "./lib/browser/interceptors/XMLHttpRequest/index.js",
        "import": "./lib/browser/interceptors/XMLHttpRequest/index.mjs",
        "default": "./lib/browser/interceptors/XMLHttpRequest/index.js"
      },
      "types": "./lib/node/interceptors/XMLHttpRequest/index.d.ts",
      "require": "./lib/node/interceptors/XMLHttpRequest/index.js",
      "import": "./lib/node/interceptors/XMLHttpRequest/index.mjs",
      "default": "./lib/node/interceptors/XMLHttpRequest/index.js"
    },
    "./fetch": {
      "browser": {
        "types": "./lib/browser/interceptors/fetch/index.d.ts",
        "require": "./lib/browser/interceptors/fetch/index.js",
        "import": "./lib/browser/interceptors/fetch/index.mjs",
        "default": "./lib/browser/interceptors/fetch/index.js"
      },
      "types": "./lib/node/interceptors/fetch/index.d.ts",
      "require": "./lib/node/interceptors/fetch/index.js",
      "import": "./lib/node/interceptors/fetch/index.mjs",
      "default": "./lib/node/interceptors/fetch/index.js"
    },
    "./RemoteHttpInterceptor": {
      "browser": null,
      "types": "./lib/node/RemoteHttpInterceptor.d.ts",
      "require": "./lib/node/RemoteHttpInterceptor.js",
      "import": "./lib/node/RemoteHttpInterceptor.mjs",
      "default": "./lib/node/RemoteHttpInterceptor.js"
    },
    "./presets/node": {
      "browser": null,
      "types": "./lib/node/presets/node.d.ts",
      "require": "./lib/node/presets/node.js",
      "import": "./lib/node/presets/node.mjs",
      "default": "./lib/node/presets/node.js"
    },
    "./presets/browser": {
      "browser": {
        "types": "./lib/browser/presets/browser.d.ts",
        "require": "./lib/browser/presets/browser.js",
        "import": "./lib/browser/presets/browser.mjs",
        "default": "./lib/browser/presets/browser.js"
      }
    }
  },
```

**Comment by steve-zhu-sh** [comment]
Hi @kettanaito, when you have the chance could you take a look at the comments above?

**Comment by kettanaito** [comment]
Hey, @steve-zhu-sh. Thanks for describing the issue and even proposing a fix!

I'm wondering if the error here isn't in having `src/index.ts` in the browser build's entries? I'd rather the browser bundle _reused_ the root-level exports as those are meant to be environment-agnostic (if they aren't, that's a bug). 

At the same time, I don't have the time to reorganize the entire exports right now so your solution should work as well. Let me give it a try.

**Comment by kettanaito** [comment]
## Released: v0.25.12 🎉

This has been released in v0.25.12!

- 📄 [**Release notes**](https://github.com/mswjs/interceptors/releases/tag/v0.25.12)
- 📦 [npm package](https://www.npmjs.com/package/@mswjs/interceptors/v/0.25.12)

Make sure to always update to the latest version (`npm i @mswjs/interceptors@latest`) to get the newest features and bug fixes.

---

_Predictable release automation by [@ossjs/release](https://github.com/ossjs/release)_.

**Comment by steve-zhu-sh** [comment]
> Hey, @steve-zhu-sh. Thanks for describing the issue and even proposing a fix!
> 
> I'm wondering if the error here isn't in having `src/index.ts` in the browser build's entries? I'd rather the browser bundle _reused_ the root-level exports as those are meant to be environment-agnostic (if they aren't, that's a bug).
> 
> At the same time, I don't have the time to reorganize the entire exports right now so your solution should work as well. Let me give it a try.

I think with the build process (and export contract) you've selected, that isn't possible. Even if the root-level exports that you've implemented in TypeScript pre build step are environment-agnostic, since you've split the build into to different environments, it seems there could be things that potentially break that environment agnostic implementation. Since browser and node may have different levels of javascript support, as a contrived example, if you build node for es6 support, but the browser for es3, there could be different levels of transpilation required for the two builds even for the shared environment agnostic modules.

I think some other library authors may choose to build to the most recent version of ecmascript and let the library consumer handle transpilation. Or at least support node and let the consumer handle browser transpilation since the user probably needs to handle that anyway for their own code due to varying levels of browser support for the different versions of ecmascript.

tsup config reference:
```ts
import { Options, defineConfig } from 'tsup'

const nodeConfig: Options = {
  entry: [
    './src/index.ts',
    './src/presets/node.ts',
    './src/RemoteHttpInterceptor.ts',
    './src/interceptors/ClientRequest/index.ts',
    './src/interceptors/XMLHttpRequest/index.ts',
    './src/interceptors/fetch/index.ts',
  ],
  outDir: './lib/node',
  platform: 'node',
  format: ['cjs', 'esm'],
  dts: true,
}

const browserConfig: Options = {
  entry: [
    './src/index.ts',
    './src/presets/browser.ts',
    './src/interceptors/XMLHttpRequest/index.ts',
    './src/interceptors/fetch/index.ts',
  ],
  outDir: './lib/browser',
  platform: 'browser',
  format: ['cjs', 'esm'],
  dts: true,
}

export default defineConfig([nodeConfig, browserConfig])
```
